app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -191,10 +190,12 @@ class StateFragmentPresenter @Inject constructor(
           helpIndex.indexTypeCase == HelpIndex.IndexTypeCase.SHOW_SOLUTION
         viewModel.setHintOpenedAndUnRevealedVisibility(true)
         viewModel.setHintBulbVisibility(true)
+        currentStateNameOfVisibleHint = currentStateName
```

Comment:
Is it possible to solve this in the assembler? Otherwise this needs to be duplicated for both states and questions.

Reply:
Yes, I tried this approach too. What I am finding difficulty here is when I am in a current state, the bulb is visible, now I move to the previous state and come back to the current state, it's a bit hard to just make the build visible. 
When coming back to the current state where prev the bulb is visible, I had to manage the last showed hint state and also start the new state like scheduling for next hints.

Reply:
Good point re: questions.

Are you still running into issues or is the current solution working?